File: GreenDiamond/GreenDiamond/Tests/Test0001.cpp
#include "all.h"

void Test0001(void)
{
	const int MAX = 45;
	int i = 0;

	SetCurtain();
	FreezeInput();

	for(; ; )
	{
		if(1 <= GetInput(INP_A))
		{
			break;
		}
		if(GetInput(INP_DIR_4) == 1)
		{
			i--;
		}
		if(GetInput(INP_DIR_6) == 1)
		{
			i++;
		}
		i = (i + MAX) % MAX;

		DrawCurtain();

		SetPrint();
		Print_x(xcout("%d", i));

		DrawCenter(D_PLAYER_STAND_1_00 + i | DTP, SCREEN_CENTER_X, SCREEN_CENTER_Y);
		EachFrame();
	}
	FreezeInput();
}

File: GreenDiamond/GreenDiamond/Tests/Test0001.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Charlotte.Common;

namespace Charlotte.Tests
{
	public class Test0001
	{
		public void Test01()
		{
			GameCurtain.SetCurtain();
			GameEngine.FreezeInput();

			foreach (GameScene scene in GameSceneUtils.Create(600))
			{
				if (GameInput.A.GetInput() == 1)
				{
					break;
				}

				GameCurtain.DrawCurtain();

				GameDraw.DrawBegin(GameGround.GeneralResource.Dummy, GameConsts.Screen_W / 2.0, GameConsts.Screen_H / 2.0);
				GameDraw.DrawZoom(5.0);
				GameDraw.DrawRotate(scene.Rate * 5.0);
				GameDraw.DrawEnd();

				GameEngine.EachFrame();
			}

			GameEngine.FreezeInput();
		}
	}
}
